galaxy_ng/app/api/urls.py

```diff
@@ -1,16 +1,43 @@
+from django.conf import settings
 from django.urls import include, path
 
 from . import views
 from .ui import urls as ui_urls
 from .v3 import urls as v3_urls
+from .v3 import viewsets as v3_viewsets
 
+DEFAULT_DISTRIBUTION_BASE_PATH = settings.GALAXY_API_DEFAULT_DISTRIBUTION_BASE_PATH.strip('/')
 
 app_name = "api"
 urlpatterns = [
-    path("", views.ApiRootView.as_view(), name="root"),
-    path("v3/", include(v3_urls)),
     path("v3/_ui/", include(ui_urls)),
 
+    # A set of galaxy v3 API urls per Distribution.base_path
+    # which is essentially per Repository
+    path("content/<str:path>/v3/",
+         include(v3_urls, namespace='per-distribution')),
```

Comment:
Naming. For example it will be used as URL name `api:v3:per-distribution:artifact-upload`.
We should use either this namespace or the default namespace for reverse URLs. I'd say that `content/<path>` namespace will be a default one. Therefore separate namespace maybe not needed for it at all.
Otherwise it's preferable to use better naming. Something like `content` is fine.

URL's that are not intended to be reversed by name may not have a name at all. Therefore it there are some URL's that are just aliases, please remove names that won't be used ever.

Reply:
Semi-related note: The current <str:path> will likely need to change to <path:path> in the future, where the parameter could be a nested path. ie, path will be 'some/path/to/where/distros/live/partnerco' instead of a single string like 'partnerco'.  By default pulp can use a nested path here but this pr doesn't currently. There is a wip branch that changes that and uses re_path() and a regex for matching the urls.